modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,881 @@
+
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+
+        return new Object[][]{
+                {id, NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, "id", null, groupOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateGroupWithPUTBadRequestException")
+    public void testUpdateGroupWithPUTBadRequestException(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectOLDGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group groupOld = (Group) objectOLDGroup;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT + "/" + GROUP_ID);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        when(userManager.getGroup(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(groupOld);
+        when(userManager.updateGroup(anyObject(), anyObject(), anyObject())).thenThrow(BadRequestException.class);
+        SCIMResponse scimResponse = groupResourceManager.updateWithPUT(id, scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_BAD_REQUEST);
+    }
+
+    @DataProvider(name = "dataForTestUpdateGroupWithPUTNotImplementedException")
+    public Object[][] dataToTestUpdateGroupWithPUTNotImplementedException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+
+        return new Object[][]{
+                {id, NEW_GROUP_SCIM_OBJECT_STRING_UPDATED, "id", null, groupOld}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateGroupWithPUTNotImplementedException")
+    public void testUpdateGroupWithPUTNotImplementedException(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object objectOLDGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group groupOld = (Group) objectOLDGroup;
```

Comment:
casting is not required if we directly pass the param with type as `Group` instead of passing it as an `Object`.